SDET Задание 2

oINCLUDEo opened the pull request.

oINCLUDEo committed:
First Version

Added 5 parallel API Tests(GetByID, GetAll, DeleteByID, Create), Random Data generating in DataGenerator, Specification in BaseRequests.

oINCLUDEo committed:
Fixed Parallel tests, random data for tests

added separate Entities for tests

oINCLUDEo committed:
fixed warnings(delete trash)

oINCLUDEo committed:
updated readme.md

oINCLUDEo committed:
updated readme.md

oINCLUDEo committed:
Update readme.md

oINCLUDEo committed:
added run_tests.yml for CI/CD

oINCLUDEo committed:
Merge remote-tracking branch 'origin/dev' into dev

oINCLUDEo committed:
Better Allure Steps and CI/CD with API Docker

oINCLUDEo committed:
test

bondarenkokate73 commented on src/test/java/dto/Entity.java:
```diff
@@ -0,0 +1,24 @@
+    private String title;
+    private boolean verified;
+    private Addition addition;
+    private int[] important_numbers;
```
Нарушение правил именования переменных

bondarenkokate73 commented on src/test/java/helpers/EntityHelper.java:
```diff
@@ -0,0 +1,111 @@
+    public static String createEntity() {
+        Entity entity = generateRandomEntity();
+
+        String entityId = BaseRequests.post("/create", entity)
```
Урлы лучше вынести в отдельные переменные или в проперти файл

bondarenkokate73 commented on src/test/java/tests/APITests.java:
```diff
@@ -0,0 +1,118 @@
+    public void createEntityTest() {
+        String entityId = EntityHelper.createEntity();
+
+        Entity createdEntity = BaseRequests.get("/get/" + entityId)
+                .then()
+                .statusCode(200)
+                .extract()
+                .as(Entity.class);
```
Запросы к апи с тестах - плохая практика, вынеси в хелперы

oINCLUDEo committed:
fixed revision

URLS now in variable, requests to api not in tests(now in EntityHelper), updates name for variable

oINCLUDEo committed:
Add Negative Tests

added tests for non-existent requests

oINCLUDEo committed:
Add getAll Test with params

Add getAllEntitiesWithRandomParamsTest with generating params(title, verified, page, perPage)

oINCLUDEo committed:
update run_tests.yml

now waiting api not unlimited

oINCLUDEo committed:
Restructure Tests

now positive tests in PositiveAPITests, negative tests in NegativeAPITests, and main BaseAPITests
